Runtimes/Supplemental/cmake/modules/FindSwiftOverlay.cmake

```
@@ -0,0 +1,240 @@
+  list(APPEND OVERLAY_TARGET_NAMES "swiftCRT")
+
+  list(APPEND swiftWinSDK_INCLUDE_DIR_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/windows"
+    "$ENV{SDKROOT}/usr/lib/swift/windows")
+  list(APPEND swiftWinSDK_LIBRARY_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/${${PROJECT_NAME}_PLATFORM_SUBDIR}/${${PROJECT_NAME}_ARCH_SUBDIR}"
+    "${Swift_SDKROOT}/usr/lib/swift"
+    "$ENV{SDKROOT}/usr/lib/swift/${${PROJECT_NAME}_PLATFORM_SUBDIR}/${${PROJECT_NAME}_ARCH_SUBDIR}"
+    "$ENV{SDKROOT}/usr/lib/swift")
+
+  if(SwiftOverlay_USE_STATIC_LIBS)
+    list(APPEND swiftCRT_NAMES libswiftWinSDK.lib)
+  else()
+    list(APPEND swiftCRT_NAMES swiftWinSDK.lib)
+  endif()
+  list(APPEND swiftWinSDK_MODULE_NAME "WinSDK.swiftmodule")
+
+  list(APPEND swiftCRT_INCLUDE_DIR_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/windows"
+    "$ENV{SDKROOT}/usr/lib/swift/windows")
+  list(APPEND swiftCRT_LIBRARY_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/${${PROJECT_NAME}_PLATFORM_SUBDIR}/${${PROJECT_NAME}_ARCH_SUBDIR}"
+    "${Swift_SDKROOT}/usr/lib/swift"
+    "$ENV{SDKROOT}/usr/lib/swift/${${PROJECT_NAME}_PLATFORM_SUBDIR}/${${PROJECT_NAME}_ARCH_SUBDIR}"
+    "$ENV{SDKROOT}/usr/lib/swift")
+
+ if(SwiftOverlay_USE_STATIC_LIBS)
+    list(APPEND swiftCRT_NAMES libswiftCRT.lib)
+  else()
+    list(APPEND swiftCRT_NAMES swiftCRT.lib)
+  endif()
+  set(swiftCRT_MODULE_NAME "CRT.swiftmodule")
+elseif(ANDROID)
+  list(APPEND OVERLAY_TARGET_NAMES "swiftAndroid")
+
+  if (SwiftOverlay_USE_STATIC_LIBS)
+    list(APPEND swiftAndroid_INCLUDE_DIR_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift_static/android"
+      "$ENV{SDKROOT}/usr/lib/swift_static/android")
+    list(APPEND swiftAndroid_LIBRARY_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift_static/android/${${PROJECT_NAME}_ARCH_SUBDIR}"
+      "${Swift_SDKROOT}/usr/lib/swift_static"
+      "$ENV{SDKROOT}/usr/lib/swift_static/android/${${PROJECT_NAME}_ARCH_SUBDIR}"
+      "$ENV{SDKROOT}/usr/lib/swift_static")
+    list(APPEND swiftAndroid_NAMES libswiftAndroid.a)
+  else()
+    list(APPEND swiftAndroid_INCLUDE_DIR_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift/android"
+      "$ENV{SDKROOT}/usr/lib/swift/android")
+    list(APPEND swiftAndroid_LIBRARY_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift/android/${${PROJECT_NAME}_ARCH_SUBDIR}"
+      "${Swift_SDKROOT}/usr/lib/swift"
+      "$ENV{SDKROOT}/usr/lib/swift/android/${${PROJECT_NAME}_ARCH_SUBDIR}"
+      "$ENV{SDKROOT}/usr/lib/swift")
+    list(APPEND swiftAndroid_NAMES libswiftAndroid.so)
+  endif()
+  set(swiftAndroid_MODULE_NAME "Android.swiftmodule")
+else()
+  message(FATAL_ERROR "FindSwiftOverlay.cmake module search not implemented for targeted platform\n"
+  " Build the Overlays for your platform and set the appropriate `<OverlayTarget>_DIR` variable to"
+  " the directory containing <OverlayTarget>Config.cmake\n")
+endif()
+
+# Setup SwiftOverlay interface library and link it against the explicit
+# overlay targets
+add_library(SwiftOverlay INTERFACE)
```

(Can be done in a follow up PR) I would use the same camel case convention we use for the other targets -- that is, start the name with a lower case letter

```suggestion
add_library(swiftOverlay INTERFACE)
```

---

@edymtt I' would probably tackle this in a another PR, but a bit confused, I see the target in the Overlay project is called `SwiftOverlay` and the `FindSwiftCore` defines `SwiftCore` so I was going off that. Might be overlooking something